### ISSUE TITLE
State is not serializable

### ISSUE THREAD

**Issue by sapostq** [cisco/mlspp#324]
tls::marshal(state) fails because State is not serializable.  We need a way to persist latest state of the client to be able to derive the new state.  In memory caching cannot be relied on.

**Comment by bifurcation** [comment]
You are correct that this is an issue.  I think that a PR to make State serializable would be fairly compact -- you would just need to add a `TLS_SERIALIZABLE` macro invocation to the State definition, and maybe make some of the constituent types also serializable.  I would be happy to review a PR, this would be good functionality to have.
